### DISCUSSION TITLE
How to draw a large-scale network graph using networkx（Occlusion issue）

### DISCUSSION THREAD

**Discussion by catbao** [networkx/networkx#5961] [Q&A]
There is a social network graph with more than 6000 vertices and tens of thousands of edges. It is drawn directly by Python networkx as shown in the figure. **How to make her look better and clearer, and make interactive information more obvious?**

![](https://i.stack.imgur.com/otSgm.png)

And what I have done is try to **decrease all those transparencies or make the line thinner.** But it looks like the result is still disorganized. Because my screen is fixed, no matter what I do, the lines are very dense just like the pic before.

My little thinking is **I don't try to make all lines clear but to make some lines clear**. It may be like, increasing the opacity of some lines which are more important or more prominent structures, decreasing the opacity of other lines. How about this approach？

Is there anyone who can give me some methods, or just a little advice or possible ideas？

**Comment by rossbar** [comment]
Visualizing graphs with many nodes/edges is quite challenging.

Without going into too much detail, I think the one thing you could try that would make the biggest impact for this particular visualization is to use a node layout that reflects some of the underlying structure of the graph. It looks to me like the image above was created with `random_layout`, i.e. the node positions are determined at random. Depending on the structure of the graph itself, you may have better luck with one of the force-directed layout functions such as `spring_layout`. A layout like this will position nodes based on their connections to one another, which can help visualization reflect underlying structure in the network. Of course, this will only get you so far - if each node has many connections then you can still end up with a very cluttered image (sometime referred to as a "hairball"). In that case maybe `circular_layout` will be better - see e.g. [the Roget example from the gallery](https://networkx.org/documentation/latest/auto_examples/graph/plot_roget.html#sphx-glr-auto-examples-graph-plot-roget-py).

Ultimately you'll need to find a layout that highlights what you want to visually communicate about your graph.

**Reply by catbao** [reply]
Thank you for your anwser! I think its very useful. But I have tried and I feel that when there are many nodes and edges, all the layout effects may not have a good effect and will get a "hairball", even if it uses a suitable layout. Maybe transparency also can help with it?

**Reply by rossbar** [reply]
You can certainly reduce the transparency of nodes/edges to see if that helps. You can specify transparency either by the `alpha` keyword argument, or by specifying color as an rgba value (there are slight differences in how this is done for drawing nodes/edges - see the docs for details).
